src/common/parse-double.c

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <locale.h>
+#include <stdlib.h>
+#include <string.h>
+#include <wlr/util/log.h>
+#include "common/mem.h"
+#include "common/parse-double.h"
+
+struct separator_loc {
+	char *chr;
+	bool multiple;
+};
+
+struct converted_double {
+	double value;
+	bool valid;
+};
+
+static struct separator_loc
+find_separator(const char *str, int c)
+{
+	struct separator_loc loc = { 0 };
+
+	loc.chr = strchr(str, c);
+	if (!loc.chr) {
+		return loc;
+	}
+
+	if (strchr(loc.chr + 1, c)) {
+		loc.multiple = true;
+	}
+
+	return loc;
+}
+
+static struct converted_double
+convert_double(const char *str)
+{
+	struct converted_double result = {
+		.value = 0,
+		.valid = true,
+	};
+
+	char *eptr = NULL;
+
+	errno = 0;
+	result.value = strtod(str, &eptr);
+
+	if (errno) {
+		wlr_log(WLR_ERROR, "double value is out of range");
+		result.valid = false;
+	}
+
+	if (*eptr) {
+		wlr_log(WLR_ERROR, "double value contains trailing garbage");
+		result.valid = false;
+	}
+
+	return result;
+}
+
+bool
+parse_double(const char *str, double *val)
```

Comment:
To keep it consistent with `common/parse-bool.h` I'd suggest `set_double()` here.